src/helpers/environment.ts

```diff
@@ -8,5 +10,20 @@ export const replaceEnvironmentVariables = (str: string) => {
   Object.keys(secrets).forEach((key) => {
     str = str.replace(`$${key}`, secrets[key] || `$${key}`);
   });
-  return str;
+  return substituteRandomNumbers(str);
 };
+
+const substituteRandomNumbers = (str: string) => {
+  if (str.includes(DYNAMIC_RANDOM_NUMBER)) {
+    const min = parseInt(process.env.RANDOM_MIN || RANDOM_MIN_DEFAULT);
+    const max = parseInt(process.env.RANDOM_MAX || RANDOM_MAX_DEFAULT);
+    while (str.includes(DYNAMIC_RANDOM_NUMBER)) {
```

Comment:
What do you think of using  [`String.prototype.replaceAll()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll) instead of the `while` loop here? It's supported Node.js v15 onwards and GitHub Actions uses Node.js v16.

Reply:
Indeed! 👍